### ISSUE TITLE
Running Method 2 QE from Journal Page places tokens on wrong map note.

### ISSUE THREAD

**Issue by Bradfordly** [spetzel2020/quick-encounters#122]
**Foundry version: 10.290**
**QE version: 1.1.2**

I'm saving Quick Encounters via Method 2. I have setup pages on my Journal with encounters on several of these pages. When I place the map notes which link to the Journal pages with the quick encounters and run the encounter, tokens are placed around the wrong map note.

I've been able to identify a pattern. When I try to run the QE, the tokens get placed around the map note which links to a journal page whose page number is nearest to 0.

Steps to reproduce:

1. Create journal object
2. Create pages in the journal object
3. Edit pages, add actor links, and save the pages
4. Create a new scene
5. Activate the scene
6. Drag a journal page to create a map note on the scene
7. Repeat step 6 with several unique pages and QEs
8. Double click one of the map note to open the journal page
9. Click "Run QE" from the prompt which opens when I open the journal page

Following these steps, the tokens are placed onto the scene as described above, and the following log output is observed:

```
Error: You are accessing PlaceableObject#data which is no longer used and instead the Document class should be referenced directly as PlaceableObject#document.
Deprecated since Version 10
Backwards-compatible support will be removed in Version 12
    logCompatibilityWarning https://<game-url>/scripts/commons.js:1692
    get https://<game-url>/scripts/foundry.js:24741
    run https://<game-url>/modules/quick-encounters/1.1.2/modules/QuickEncounter.js:1117
    activateListeners https://<game-url>/modules/quick-encounters/1.1.2/modules/QESheet.js:125
    jQuery 2
        dispatch
        handle
```

```
Uncaught (in promise) TypeError: this.target is undefined
[Detected 1 package: quick-encounters]
    _refreshTarget https://<game-url>/scripts/foundry.js:46044
    _refresh https://<game-url>/scripts/foundry.js:45937
    refresh https://<game-url>/scripts/foundry.js:25134
    _onControl https://<game-url>/scripts/foundry.js:46978
    control https://<game-url>/scripts/foundry.js:25229
    createCombat https://<game-url>/modules/quick-encounters/1.1.2/modules/QuickEncounter.js:1518
    run https://<game-url>/modules/quick-encounters/1.1.2/modules/QuickEncounter.js:1174
[foundry.](https://<game-url>/scripts/foundry.js)
```

**Comment by spetzel2020** [comment]
Thanks for the report; you have identified a problem I hadn't anticipated:
I have currently assumed that the single map note for the Journal Entry is
used for all Journal Pages.

However, I can see why it should work as you describe. I'm away from my
computer for about a week but will look at it then.

**Comment by Bradfordly** [comment]
Appreciate the response! I'll have a little time to spare this evening - if there's a specific method you could point me at... I might be able to contribute a fix to help you out if you'd like.

**Comment by spetzel2020** [comment]
Sorry for the delay; I have been away from QE for several months but am looking at this now.

**Comment by spetzel2020** [comment]
Things that should happen:
1. When you drag the Journal Entry Page to the Map, it should be intercepted in the same way that EncounterNote.js#346 does in the `Hooks.on(`dropCanvasData``
2. Currently, when you double click the Map Note, you get the correct Journal Entry Page and Quick Encounter popped up, and the correct Quick Encounter run. But the Map Note used is incorrect.

**Comment by spetzel2020** [comment]
In `run()`, `line 1093` we specifically record the parent JE rather than the actual JE Page on the assumption that only the main JE is on the Scene.

Instead we should find either the Map Note for this JE or only use the parent position if the child Map Note doesn't exist.

**Comment by spetzel2020** [comment]
Refactored `run()` to not check for `JournalEntryPage`; instead added `findMapNoteForJE()` which we call first for the `JournalEntry` and then (if no Map Note is found) again with the parent JE if this is a `JournalEntryPage`

This allows you to drag only some of the `JournalEntryPage`s as map notes; others will use the JE one (or prompt).

**Comment by spetzel2020** [comment]
Released in 1.1.4
